### PR TITLE
Usability updates to onnx parsing

### PR DESCRIPTION
**Pull request by pfultz2**
This will now throw an error on missing operator and there is a flag to enable the older behaviour.

There is also a flag to print the program on parsing error.

### CODE REVIEW

**Comment by kahmed10** [comment]
Do we plan on opening a separate PR for the tf parsing changes?

**Comment by pfultz2** [comment]
> Do we plan on opening a separate PR for the tf parsing changes?

We can.

File: src/py/migraphx_py.cpp [hunk 1]
```diff
@@ -181,12 +181,20 @@ PYBIND11_MODULE(migraphx, m)
...
+                                          {.batch_size             = batch_size,
+                                           .skip_unknown_operators = skip_unknown_operators,
+                                           .print_program_on_error = print_program_on_error});
           },
           "Parse onnx file",
           py::arg("filename"),
-          py::arg("batch_size") = 1);
+          py::arg("batch_size")             = 1,
+          py::arg("skip_unknown_operators") = false,
+          py::arg("print_program_on_error") = false);
```
Comment (kahmed10) [inline src/py/migraphx_py.cpp:197]: Do we want the default behavior in the python API to also print_program_on_error like the driver?

**Comment by pfultz2** [comment]
The failure is from code coverage, but I dont think we can recover. Its mainly missing the try/catch coverage.

File: src/py/migraphx_py.cpp (hunk 1 above)
Comment (pfultz2) [reply src/py/migraphx_py.cpp:197]: I dont think so, as this can be used by applications.

File: src/onnx/onnx.cpp [hunk 2]
```diff
@@ -2112,21 +2116,25 @@ template <class... Ts>
...
-#ifndef NDEBUG
-    // Log the program when it can't be parsed
-    try
+    parser.batch_size             = options.batch_size;
+    parser.skip_unknown_operators = options.skip_unknown_operators;
+    if(options.print_program_on_error)
     {
-        parser.parse_from(std::forward<Ts>(xs)...);
+        // Log the program when it can't be parsed
+        try
```
Comment (kahmed10) [inline src/onnx/onnx.cpp:2124]: One way to get better coverage could be to call parse_onnx() in onnx_test.cpp with print_program_on_error=true.

File: test/onnx/onnx_test.cpp [hunk 3]
```diff
@@ -1619,6 +1621,11 @@ TEST_CASE(unknown_test)
     EXPECT(p == prog);
 }
 
+TEST_CASE(unknown_test_throw)
+{
+    EXPECT(test::throws([&] { migraphx::parse_onnx("unknown_test.onnx"); }));
```
Comment (kahmed10) [inline test/onnx/onnx_test.cpp:1626]: you can repeat this call but add print_program_on_error=true to the options.

**Pull request merged**